src/components/AttachmentModal.tsx

```diff
@@ -359,62 +389,60 @@ function AttachmentModal(props) {
     }, []);
 
     useEffect(() => {
-        setSource(props.source);
-    }, [props.source]);
+        setSourceState(source);
+    }, [source]);
 
     useEffect(() => {
-        setIsAuthTokenRequired(props.isAuthTokenRequired);
-    }, [props.isAuthTokenRequired]);
+        setIsAuthTokenRequiredState(isAuthTokenRequired);
+    }, [isAuthTokenRequired]);
 
-    const sourceForAttachmentView = source || props.source;
+    const sourceForAttachmentView = sourceState ?? source;
```

Comment:
`sourceState` can be empty string which will case problems
```suggestion
    const sourceForAttachmentView = sourceState || source;
```